### PR TITLE
Added component specs for ConsumePipeline

### PR DESCRIPTION
**Pull request by jpjerkins**

### CODE REVIEW

File: src/NetStream.Specs/Specifications/Component/ConsumePipelineSpecs.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        [Subject("AppendStep")]
+        class when_appending_multiple_steps
+        {
+            static ConsumePipeline<string, string> _pipeline;
+            static string _log = "";
+            static TestHandleStep<string, string> _handle1;
+            static TestHandleStep<string, string> _handle2;
+            static TestHandleStep<string, string> _handle3;
+
+            private Establish context = () =>
```
Comment (joelbrinkley) [inline src/NetStream.Specs/Specifications/Component/ConsumePipelineSpecs.cs]: You are showing your privates again :P

File: src/NetStreams/Internal/ConsumePipeline.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 namespace NetStreams.Internal
 {
-    internal class ConsumePipeline<TKey, TMessage> : IConsumePipeline<TKey, TMessage>
+    public class ConsumePipeline<TKey, TMessage> : IConsumePipeline<TKey, TMessage>
```
Comment (joelbrinkley) [inline src/NetStreams/Internal/ConsumePipeline.cs:6]: If we make this public then we need to remove it from the internal name space.

File: src/NetStream.Specs/Specifications/Component/ConsumePipelineSpecs.cs (hunk 1 above)
Comment (jpjerkins) [reply src/NetStream.Specs/Specifications/Component/ConsumePipelineSpecs.cs]: I believe this to now be resolved.

File: src/NetStreams/Internal/ConsumePipeline.cs (hunk 2 above)
Comment (jpjerkins) [reply src/NetStreams/Internal/ConsumePipeline.cs:6]: Moved the class out of the Internal namespace.

**Pull request merged**